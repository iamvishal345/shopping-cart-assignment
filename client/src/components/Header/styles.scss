@import "assets/scss/_breakpoints.scss";

@mixin nav-list {
  list-style: none;
  display: flex;
  margin: 0;
}

.header-container {
  display: flex;
  justify-content: space-between;
  padding: 0px 10rem;
  box-shadow: 0px 1px 9px 2px #ededed;
  @include lg {
    padding: 0 4rem;
  }
  @include md {
    padding: 0 2rem;
  }
  @include xm {
    padding: 0;
    padding-left: 4px;
  }
  .left {
    display: flex;
    align-items: center;
    nav {
      align-self: flex-end;
      @include xm {
        display: none;
      }
      .nav-list {
        @include nav-list;
        margin-bottom: 8px;
        .nav-item {
          padding: 8px;
          a {
            text-decoration: none;
            font-size: 14px;
            color: #79716f;
            font-weight: 500;
          }
        }
      }
    }
  }
  .right {
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    .login-signup-nav {
      @include nav-list;
      @include xm {
        display: none;
      }
      .nav-item {
        padding: 6px;
        a {
          text-decoration: none;
          font-size: 12px;
          color: #000000;
        }
      }
    }
  }

  .active {
    color: #d10157 !important;
  }
}
