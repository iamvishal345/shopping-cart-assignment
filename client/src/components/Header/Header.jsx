import React from "react";
import "./styles.scss";
import { NavLink } from "react-router-dom";

import { Logo } from "components/Atoms/Logo/Logo";
import { Cart } from "components/Atoms/Cart/Cart";

export const Header = () => (
  <div className="header-container">
    <div className="left">
      <Logo>Sabka Bazaar</Logo>
      <nav role="navigation" aria-label="Primary menu">
        <ul className="nav-list">
          <li className="nav-item">
            <NavLink exact activeClassName="active" to="/">
              Home
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink activeClassName="active" to="/products">
              Products
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
    <div className="right">
      <nav role="navigation" aria-label="Secondary menu">
        <ul className="login-signup-nav">
          <li className="nav-item">
            <NavLink activeClassName="active" to="/signIn">
              SignIn
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink activeClassName="active" to="/register">
              Register
            </NavLink>
          </li>
        </ul>
      </nav>
      <Cart />
    </div>
  </div>
);
