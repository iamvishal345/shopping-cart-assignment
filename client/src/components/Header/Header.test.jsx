import React from "react";
import { shallow } from "enzyme";
import { Header } from "./Header";
// import { createMemoryHistory } from "history";
// import { NavLink } from "react-router-dom";

const wrapped = shallow(<Header />);
// const history = createMemoryHistory();
describe("Header", () => {
  it("should match the snapshot", () => {
    expect(wrapped).toMatchSnapshot();
  });

  // it("should naviagate to product page", () => {
  //   wrapped.find(NavLink).at(2).simulate("click", {});
  //   expect(history.location.pathname).toBe("/products");
  // });
});
