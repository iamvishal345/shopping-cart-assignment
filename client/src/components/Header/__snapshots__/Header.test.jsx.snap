// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div
  className="header-container"
>
  <div
    className="left"
  >
    <Logo>
      Sabka Bazaar
    </Logo>
    <nav
      aria-label="Primary menu"
      role="navigation"
    >
      <ul
        className="nav-list"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            to="/"
          >
            Home
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            to="/products"
          >
            Products
          </NavLink>
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="right"
  >
    <nav
      aria-label="Secondary menu"
      role="navigation"
    >
      <ul
        className="login-signup-nav"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            to="/signIn"
          >
            SignIn
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            to="/register"
          >
            Register
          </NavLink>
        </li>
      </ul>
    </nav>
    <Cart />
  </div>
</div>
`;
